rule/string-regex.go

```diff
@@ -0,0 +1,233 @@
+package rule
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"regexp"
+	"strconv"
+
+	"github.com/mgechev/revive/lint"
+)
+
+// #region Revive API
+
+// StringRegexRule lints strings and/or comments according to a set of regular expressions given as Arguments
+type StringRegexRule struct{}
+
+// Apply applies the rule to the given file.
+func (r *StringRegexRule) Apply(file *lint.File, arguments lint.Arguments) []lint.Failure {
+	var failures []lint.Failure
+
+	onFailure := func(failure lint.Failure) {
+		failures = append(failures, failure)
+	}
+
+	w := lintStringRegexRule{onFailure: onFailure}
+	w.parseArguments(arguments)
+	ast.Walk(w, file.AST)
+
+	return failures
+}
+
+func (r *StringRegexRule) Name() string {
+	return "string-regex"
+}
+
+// #endregion
+
+// #region Internal structure
+
+type lintStringRegexRule struct {
+	onFailure func(lint.Failure)
+
+	rules              []stringRegexSubrule
+	stringDeclarations map[string]string
+}
+
+type stringRegexSubrule struct {
+	Parent       *lintStringRegexRule
+	Scope        stringRegexSubruleScope
+	Regexp       *regexp.Regexp
+	ErrorMessage string
+}
+
+type stringRegexSubruleScope struct {
+	Func     string // Function name the rule is scoped to
+	Argument int    // (optional) Which argument in calls to the function is checked against the rule (the first argument is checked by default)
+	Field    string // (optional) If the argument to be checked is a struct, which member of the struct is checked against the rule (top level members only)
+}
+
+var parseStringRegexScope = regexp.MustCompile("^([A-Za-z][A-Za-z0-9]+)(?:\\[([0-9]+)\\])?(?:\\.([A-Za-z]+))?$")
+
+// #endregion
+
+// #region Argument parsing
+
+func (w *lintStringRegexRule) parseArguments(arguments lint.Arguments) {
+	for i, argument := range arguments {
+		scope, regex, errorMessage := w.parseArgument(argument, i)
+		w.rules = append(w.rules, stringRegexSubrule{
+			Parent:       w,
+			Scope:        scope,
+			Regexp:       regex,
+			ErrorMessage: errorMessage,
+		})
+	}
+}
+
+func (w lintStringRegexRule) parseArgument(argument interface{}, ruleNum int) (scope stringRegexSubruleScope, regex *regexp.Regexp, errorMessage string) {
```

Comment:
the complexity of the functions related with the parsing of arguments might deserve a dedicated set of unitary test

Reply:
I agree that a set of tests for argument parsing would be good. I don't have the time to implement this tonight, but I can get it done tomorrow.